app/react/kubernetes/applications/components/RedeployGitAppForm/RedeployGitAppForm.tsx

```diff
@@ -0,0 +1,362 @@
+    function automaticUpdatesLabel(
+      repositoryAutomaticUpdates: boolean,
+      repositoryMechanism: AutoUpdateMechanism
+    ) {
+      switch (repositoryAutomaticUpdates && repositoryMechanism) {
+        case RepositoryMechanismTypes.INTERVAL:
+          return 'polling';
+        case RepositoryMechanismTypes.WEBHOOK:
+          return 'webhook';
+        default:
+          return 'off';
+      }
+    }
+  }
+}
+
+function RedeployGitAppInnerForm({
+  stack,
+  namespaceName,
+  webhookId,
+}: {
+  stack: Stack;
+  namespaceName: string;
+  webhookId: string;
+}) {
+  const router = useRouter();
+  const environmentId = useEnvironmentId();
+  const redeployKubeGitStackMutation = useRedeployKubeGitStackMutation(
+    stack.Id,
+    environmentId
+  );
+  const [isRedeployLoading, setIsRedeployLoading] = useState(false);
+  const { trackEvent } = useAnalytics();
+  const {
+    values,
+    errors,
+    setFieldValue,
+    handleSubmit,
+    dirty,
+    isValid,
+    isSubmitting,
+    setFieldTouched,
+  } = useFormikContext<KubeAppGitFormValues>();
+
+  return (
+    <Form className="form-horizontal" onSubmit={handleSubmit}>
+      <AutoUpdateFieldset
+        value={values.autoUpdate}
+        onChange={(value: AutoUpdateModel) =>
+          setFieldValue('autoUpdate', value)
+        }
+        baseWebhookUrl={baseStackWebhookUrl()}
+        webhookId={webhookId}
+        errors={errors.autoUpdate}
+        environmentType="KUBERNETES"
+        isForcePullVisible={false}
+        webhooksDocs="https://docs.portainer.io/user/kubernetes/applications/webhooks"
+      />
+      <TimeWindowDisplay />
+      <FormSection title="Advanced configuration" isFoldable>
+        <RefField
+          value={values.repositoryReferenceName}
+          onChange={(refName) =>
+            setFieldValue('repositoryReferenceName', refName)
+          }
+          model={{
+            ...values.authentication,
+            RepositoryURL: values.repositoryURL,
+          }}
+          error={errors.repositoryReferenceName}
+          stackId={stack.Id}
+          isUrlValid
+        />
+        <AuthFieldset
+          value={values.authentication}
+          isAuthExplanationVisible
+          onChange={(value) =>
+            Object.entries(value).forEach(([key, value]) => {
+              setFieldValue(key, value);
+              // set touched after a delay to revalidate debounced username and access token fields
+              setTimeout(() => setFieldTouched(key, true), 400);
+            })
+          }
+          errors={errors.authentication}
+        />
+        <SwitchField
+          name="TLSSkipVerify"
+          label="Skip TLS verification"
+          labelClass="col-sm-3 col-lg-2"
+          tooltip="Enabling this will allow skipping TLS validation for any self-signed certificate."
+          checked={values.tlsSkipVerify}
+          onChange={onChangeTLSSkipVerify}
+        />
+      </FormSection>
+      <FormSection title="Actions">
```

Comment:
use `FormActions`?

Reply:
For migrating 'as is' the Save settings button (type=submit) is actually the second button, so it doesn't perfectly match the form actions component, which has the (type=submit) button first.

I'll check with James if I can update the order and style of these buttons to match the form actions component and other forms in portainer, so that I can use form actions.